satpy/tests/reader_tests/test_nwcsaf_nc.py

```diff
@@ -35,13 +39,17 @@ class TestNcNWCSAF(unittest.TestCase):
     """Test the NcNWCSAF reader."""
 
     @mock.patch('satpy.readers.nwcsaf_nc.unzip_file')
-    @mock.patch('satpy.readers.nwcsaf_nc.xr')
-    def setUp(self, xr_, unzip):
+    @mock.patch('satpy.readers.nwcsaf_nc.xr.open_dataset')
+    def setUp(self, xr_open_dataset, unzip):
         """Set up the test case."""
         from satpy.readers.nwcsaf_nc import NcNWCSAF
-        xr_.return_value = mock.Mock(attrs={})
+        xr_open_dataset.return_value = xr.Dataset({"nx": xr.DataArray(), "ny": xr.DataArray()},
+                                                  attrs={"source": "bla",
+                                                         "satellite_identifier": "blu"})
+        self.fake_dataset = xr_open_dataset.return_value
         unzip.return_value = ''
-        self.scn = NcNWCSAF('filename', {}, {})
+        self.filehandler_class = NcNWCSAF
+        self.scn = self.filehandler_class('filename', {}, {})
```

Comment:
Can this property be renamed to avoid confusion? I was already confused when you set `self.scn.nc` in one of the tests since that isn't something a Scene should have.

Reply:
ah yes, good poing